src/Core/Geometry/TriangleMesh.hpp

```diff
@@ -187,28 +190,52 @@ class RA_CORE_API TriangleMesh : public AbstractGeometry
     /// The handle for normals, making request faster.
     NormalAttribHandle m_normalsHandle;
 
-    /// Sets the default attribs.
-    inline void initDefaultAttribs() {
-        m_verticesHandle =
-            m_vertexAttribs.addAttrib<PointAttribHandle::value_type>( "in_position" );
-        m_normalsHandle = m_vertexAttribs.addAttrib<NormalAttribHandle::value_type>( "in_normal" );
-    }
-
-    /// Append the data of \p attr to the attribute with the same name.
-    /// \warning There is no check on the existence of *this's attribute.
-    /// \warning There is no error check on the handles attribute type.
-    template <typename T>
-    void append_attrib( Utils::AttribBase* attr ) {
-        auto h         = m_vertexAttribs.findAttrib<T>( attr->getName() );
-        auto& v0       = m_vertexAttribs.getAttrib( h ).data();
-        const auto& v1 = attr->cast<T>().data();
-        v0.insert( v0.end(), v1.cbegin(), v1.cend() );
-    }
-
     // Ease wrapper
     friend class TopologicalMesh;
 };
 
+class RA_CORE_API PointCloud : public AttribArrayGeometry
+{};
+
+class RA_CORE_API LineStrip : public AttribArrayGeometry
+{};
+
+/// Simple Mesh structure that handles indexed polygonal mesh with vertex
+/// attributes. Each face is indexed with typename T = IndexType.
+/// T is assumed to be an Eigen Vector of unsigned ints
+template <typename T>
+class RA_CORE_API IndexedGeometry : public AttribArrayGeometry
+{
+  public:
+    using IndexType          = T;
+    using IndexContainerType = AlignedStdVector<IndexType>;
+
+    inline IndexedGeometry() = default;
+    inline explicit IndexedGeometry( const IndexedGeometry<IndexType>& other );
+    inline IndexedGeometry( IndexedGeometry<IndexType>&& other );
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'IndexedGeometry < Vector2ui >' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/app/STORM/Radium-Engine/pullRequest?prid=3878283)